src/script/media/MediaDevicesHandler.ts

```diff
@@ -165,18 +189,37 @@ export class MediaDevicesHandler {
    */
   private subscribeToObservables(): void {
     this.currentDeviceId.audioInput.subscribe(mediaDeviceId => {
+      const faveAudioInput = loadValue<string[]>(FavoriteDeviceTypes.AUDIO_INPUT) ?? [];
+      const disconnected = this.previousDeviceSupport.audioInput > this.availableDevices.audioInput().length;
+      const newFaveAudioInput = this.orderFavoriteDevices(mediaDeviceId, faveAudioInput, disconnected);
+
       storeValue(MediaDeviceType.AUDIO_INPUT, mediaDeviceId);
+      storeValue(FavoriteDeviceTypes.AUDIO_INPUT, newFaveAudioInput);
     });
 
     this.currentDeviceId.audioOutput.subscribe(mediaDeviceId => {
```

Comment:
In m opinion its good to create a reusable subscription function. If you ever need to make a change to the subscription logic, you only need to do it in one place. This reduces the risk of bugs and it's clear what each subscription is doing without having to repeat the logic. If you ever need to add more device types in the future, that would be easy too.

```
private subscribeToDevice(deviceType: 'audioInput' | 'audioOutput' | 'videoInput') {
    this.currentDeviceId[deviceType].subscribe(mediaDeviceId => {
        const faveDevices = loadValue<string[]>(FavoriteDeviceTypes[deviceType.toUpperCase()]) ?? [];
        const disconnected = this.previousDeviceSupport[deviceType] > this.availableDevices[deviceType]().length;
        const newFaveDevices = this.orderFavoriteDevices(mediaDeviceId, faveDevices, disconnected);

        storeValue(MediaDeviceType[deviceType.toUpperCase()], mediaDeviceId);
        storeValue(FavoriteDeviceTypes[deviceType.toUpperCase()], newFaveDevices);
    });
}

subscribeToDevice('audioInput');
subscribeToDevice('audioOutput');
subscribeToDevice('videoInput');
```